kernel/collection/ringbuffer/ringbuffer.c

```diff
@@ -0,0 +1,78 @@
+//*****************************************************************
```

Comment:
Do we need arbitrary-size ring buffers yet?  I wonder if for now, we can use fixed-size (256-element, maybe 65536-element if needed) ringbuffers implemented under the hood as 256- (or 64k-)element arrays, and we can take advantage of unsigned integer wraparound to remove some of the complexity with dealing with the wraparound of indices?

Reply:
100% agree.  I can change the ringbuffer to fixed size.

Reply:
Issue added for this.  currently ringbuffer isn't actually used, therefore we can make it what we want.